lib/src/boilerplate_suggestors/boilerplate_utilities.dart

```diff
@@ -256,7 +258,26 @@ class ClassToMixinConverter {
       }
 
       if (node.withClause != null) {
-        yieldPatch(node.withClause.offset, node.withClause.end, '');
+        // Convert the with clause to an implements clause since the mixin
+        // is how consumers will "extend" props moving forward. These mixins
+        // are moved to the new concrete class via the `AdvancedPropsAndStateClassMigrator`,
+        // but if they are not also implemented here, the new props mixin won't have all
+        // the keys that the old class / abstract class had... which will be a breaking
+        // change for consumers without a workaround/path forward.
```

Comment:
I'm not sure I agree that we should always implement these. They shouldn't always be needed, and we shouldn't clutter the boilerplate unnecessarily for cases where it's really just the concrete props class consuming another mixin that's unrelated to the rest of the props.

The conversion from a class to mixin is already a breaking change, which is fine since in most cases we're only operating on public APIs.

Anyone using the typing outside of the component should be using the concrete props class, which will have everything mixed in, so we don't need to worry about those cases.

The only case I can think of is if a mixin is overriding/implementing a prop from an interface, which should be far less common. A less intrusive way could be to check for `@override` annotations on props, and add a FIXME saying to add missing implements clauses if necessary

Reply:
@greglittlefield-wf If we don't implement the classes that used to be mixed in - in the new mixin - then conversions like this won't work:

__Existing structure__:

```dart
// --- abstract ---

@AbstractProps()
abstract class AbstractBlockProps extends UiProps
    with LayoutPropsMixin, BlockPropsMixin
    implements BlockClassHelperMapView {}
@AbstractComponent2()
abstract class AbstractBlockComponent<T extends AbstractBlockProps> extends UiComponent2<T> with LayoutMixin<T>, BlockMixin<T> {}

// --- concrete implementation ---

@Props()
class BlockProps extends AbstractBlockProps {}
@Component2()
class BlockComponent extends AbstractBlockComponent<BlockProps> {}
```

__After codemod__:

```dart
// --- abstract ---

mixin AbstractBlockPropsMixin on UiProps {}

class AbstractBlockProps = UiProps with AbstractBlockPropsMixin, 
    LayoutPropsMixin, BlockPropsMixin
    implements BlockClassHelperMapView;

abstract class AbstractBlockComponent<T extends AbstractBlockProps> extends UiComponent2<T> with LayoutMixin<T>, BlockMixin<T> {}

// --- concrete implementation ---

mixin BlockPropsMixin on UiProps {}

class BlockProps = UiProps with BlockPropsMixin, AbstractBlockPropsMixin {}

// Analysis error here: BlockProps does not implement AbstractBlockProps
class BlockComponent extends AbstractBlockComponent<BlockProps> {}
```

The last line above is the problem. IMO - we're really just dealing with WSD at this point - and a certain level of verbosity is expected.  And in this case - adding those interfaces to the mixins is necessary to make the components work in their current structure.  We can / should certainly phase this out over time... but for the initial conversion, it would be nice for the vast majority of things to "just work" with minimal intervention required - since there will be a few things that will take A LOT of intervention no matter how good the codemods are.

Reply:
Yeah, I'm worried more about non-WSD usages, and would prefer to not have unnecessary `implements` clauses that clutter things up and potentially confuse consumers. 

Could we conditionally only add implements clauses when we're the component extends abstract components?

```dart
class BlockProps = UiProps 
    with BlockPropsMixin, AbstractBlockPropsMixin 
    implements AbstractBlockProps;
```

That way, it should only be added to the cases that need it.


---

Side note: in this case, we really shouldn't be making a concrete `AbstractBlockProps` class; it will never be extended from or used, so there's effectively no difference between mixing in or implementing something. How about we just make them "interfaces"
```dart
abstract class AbstractBlockProps implements 
    AbstractBlockPropsMixin, 
    LayoutPropsMixin,
    BlockPropsMixin,
    BlockClassHelperMapView;
```

Reply:
Implemented this in https://github.com/Workiva/over_react_codemod/pull/81/commits/995a3d225aceaaf44ecb040814db9bfe050e4218